diff --git a/CMakeLists.txt b/CMakeLists.txt
index 83e7d5b..46c36ad 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -480,6 +480,9 @@ add_library(raw ${libraw_LIB_SRCS})
 add_library(libraw::libraw ALIAS raw)
 target_compile_definitions(raw PRIVATE LIBRAW_NOTHREADS)
 
+# define LIBRAW_OWN_SWAB to use own swab implementation
+target_compile_definitions(raw PRIVATE LIBRAW_OWN_SWAB)
+
 # Flag to export library symbols
 if (WIN32)
     target_compile_definitions(raw PRIVATE LIBRAW_BUILDLIB)
@@ -540,6 +543,9 @@ set(libraw_r_LIB_SRCS ${libraw_LIB_SRCS})
 add_library(raw_r ${libraw_r_LIB_SRCS})
 add_library(libraw::libraw_r ALIAS raw_r)
 
+# define LIBRAW_OWN_SWAB to use own swab implementation
+target_compile_definitions(raw_r PRIVATE LIBRAW_OWN_SWAB)
+
 # Flag to export library symbols
 if(WIN32)
     target_compile_definitions(raw_r PRIVATE LIBRAW_BUILDLIB)
