diff --git a/CMakeLists.txt b/CMakeLists.txt
index 52439b3..1e1d009 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -497,12 +497,14 @@ install(FILES "${CMAKE_CURRENT_BINARY_DIR}/liblzmaConfig.cmake"
 # The command line tools needs this.
 check_symbol_exists(getopt_long getopt.h HAVE_GETOPT_LONG)
 
+option(BUILD_COMMAND_LINE_TOOLS "build command line tools (default on)" ON)
+
 
 #############################################################################
 # xzdec
 #############################################################################
 
-if(HAVE_GETOPT_LONG)
+if(BUILD_COMMAND_LINE_TOOLS AND HAVE_GETOPT_LONG)
     add_executable(xzdec
         src/common/sysdefs.h
         src/common/tuklib_common.h
@@ -540,7 +542,7 @@ endif()
 # xz
 #############################################################################
 
-if(NOT MSVC AND HAVE_GETOPT_LONG)
+if(BUILD_COMMAND_LINE_TOOLS AND NOT MSVC AND HAVE_GETOPT_LONG)
     add_executable(xz
         src/common/mythread.h
         src/common/sysdefs.h
